src/main/java/mods/eln/sixnode/AnalogChips.kt

```diff
@@ -685,3 +683,121 @@ class SampleAndHold : AnalogFunction() {
         nbt?.setDouble("value", value)
     }
 }
+
+class Filter: AnalogFunction() {
+    override val hasState = true
+    override val inputCount = 1
+    override val infos = I18N.tr("Lowpass filter - Passes signals with a\nfrequency lower than a certain cutoff frequency\nand attenuates signals with frequencies higher\nthan the cutoff frequency.")
+
+    internal var feedback = 2.0 * Math.PI * 5.0
+    private var output = 0.0
+
+    override fun process(inputs: Array<Double?>, deltaTime: Double): Double {
+        output = Utils.limit(output + ((inputs[0] ?: 0.0) - output) * feedback * deltaTime, 0.0, 50.0)
+        return output
+    }
+
+    override fun readFromNBT(nbt: NBTTagCompound?, str: String?) {
+        nbt?.apply {
+            feedback = getDouble("feedback")
```

Comment:
Everything on the simulation side is double. I convert it to/from float when serializing to/from client.